packages/modal/README.md

```diff
@@ -0,0 +1,85 @@
+# @rooks/use-modal
+
+### Exposes modal logic through react context, also provides useToggle hook for controlling modal from other components
+
+[![Build Status](https://travis-ci.org/imbhargav5/rooks.svg?branch=master)](https://travis-ci.org/imbhargav5/rooks) ![](https://img.shields.io/npm/v/@rooks/use-modal/latest.svg) ![](https://img.shields.io/npm/l/@rooks/use-modal.svg) ![](https://img.shields.io/bundlephobia/min/@rooks/use-modal.svg) ![](https://img.shields.io/david/imbhargav5/rooks.svg?path=packages%2Fmodal)
+
+<a href="https://spectrum.chat/rooks"><img src="https://withspectrum.github.io/badge/badge.svg" alt="Join the community on Spectrum"/></a>
+
+### Installation
+
+```
+npm install --save @rooks/use-modal
+```
+
+### Importing the hook
+
+```js
+import useModal, { ModalProvider, useToggle } from "@rooks/use-modal"
```

Comment:
We already have a hook called `useToggle`. Can we name this something else?

Reply:
Refactored so `useToggle` is not needed. `useModal` registration kinda acts as singleton so only the first call to `useModal` does the registration and other calls just use already created modal context. I think this kind of API is also more straightforward. 

Also `ModalProvider` is now exported as `useModal.Provider`.

Reply:
Also added this package `packages/rooks`.